src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -0,0 +1,369 @@
+        checkState(organizationExternalDatabaseColumns.putIfAbsent(column.id, column) == null,
+                "OrganizationExternalDatabaseColumn ${columnFQN.fullQualifiedNameAsString} already exists")
+        aclKeyReservations.reserveIdAndValidateType(column, columnFQN::getFullQualifiedNameAsString)
+
+        val columnAclKey = AclKey(orgId, column.tableId, column.id)
+        authorizationManager.setSecurableObjectType(columnAclKey, SecurableObjectType.OrganizationExternalDatabaseColumn)
+        authorizationManager.addPermission(columnAclKey, principal, EnumSet.allOf(Permission::class.java))
+
+        return column.id
+    }
+
+    /*GET*/
+    fun getOrganizationIds(): Set<UUID> {
+        val orgIds = mutableSetOf<UUID>()
+        hds.connection.createStatement().use {
+            val rs = it.executeQuery("SELECT id FROM organizations")
+            while (rs.next()) {
+                orgIds.add(organizationId(rs))
+            }
+        }
+        return orgIds
+    }
+
+    fun getOrganizationExternalDatabaseTable(tableId: UUID): OrganizationExternalDatabaseTable {
+        return organizationExternalDatabaseTables[tableId]!!
+    }
+
+    fun getOrganizationExternalDatabaseColumn(columnId: UUID): OrganizationExternalDatabaseColumn {
+        return organizationExternalDatabaseColumns[columnId]!!
+    }
+
+    /*DELETE*/
+    fun deleteOrganizationExternalDatabaseTables(orgId: UUID, tableIds: Set<UUID>) {
+        tableIds.forEach { deleteOrganizationExternalDatabaseTable(orgId, it) }
+    }
+
+    fun deleteOrganizationExternalDatabaseTable(orgId: UUID, tableId: UUID) {
+        organizationExternalDatabaseTables.remove(tableId) //TODO make this a set so we can batch delete, entry processor?
+        aclKeyReservations.release(tableId)
+
+        //delete columns that belonged to this table
+        val belongsToDeletedTable = Predicates.equal("tableId", tableId)
+        val columnsToDelete = organizationExternalDatabaseColumns.values(belongsToDeletedTable)
+        val columnIds = columnsToDelete.map { it.id }.toSet()
+        deleteOrganizationExternalDatabaseColumns(orgId, columnIds)
+
+    }
+
+    fun deleteOrganizationExternalDatabaseColumns(orgId: UUID, columnIds: Set<UUID>) {
+        columnIds.forEach { deleteOrganizationExternalDatabaseColumn(orgId, it) }
+    }
+
+    fun deleteOrganizationExternalDatabaseColumn(orgId: UUID, columnId: UUID) {
+        organizationExternalDatabaseColumns.remove(columnId) //TODO make this a set so we can batch delete, entry processor?
+        aclKeyReservations.release(columnId)
+    }
+
+    /**
+     * Deletes an organization's entire database.
+     * Is called when an organization is deleted.
+     */
+    //TODO break into further helper methods to reduce reuse of code
+    fun deleteOrganizationExternalDatabase(orgId: UUID) {
+        //remove all tables/columns within org
+        val belongsToDeletedDB = Predicates.equal("organizationId", orgId)
```

Comment:
You should use the constant `ORGANIZATION_ID_INDEX` here

Reply:
again good catch :)